### PR TITLE
Update doc index page

### PR DESCRIPTION
**Pull request by mrocklin**
I hope that this presents a bit more of a high level "why" before jumping into "how"

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -2,54 +2,101 @@
...
+
+**Dask Bags** mimic iterators, Toolz, PySpark
+
+.. code-block:: python
+
+   import dask.bag as db
+   b = db.from_filenames('2015-*-*.json.gz').map(json.loads)
+   b.pluck('name').frequencies().topk(10, lambda pair: pair[1]).compute()
+
+**Dask Imperative** mimic for loops
```
Comment (jcrist) [inline docs/source/index.rst]: Perhaps mimic -> mimics?

File: docs/source/index.rst [hunk 2]
```diff
@@ -2,54 +2,101 @@
...
+
+
+Scales from laptops to clusters
+-------------------------------
+
+Dask is convenient on a laptop.  It :doc:`installs <install>` trivially with
+``conda`` or ``pip`` and extends the size of convenient datasets from "fits in
+memory" to "fits on disk".
+
+Dask is powerful to run on a cluster of 100s of machines. It is resilient,
```
Comment (jcrist) [inline docs/source/index.rst]: "Dask is powerful to run on a cluster..." -> "dask is powerful enough to run on a cluster..."?

File: docs/source/index.rst [hunk 3]
```diff
@@ -2,54 +2,101 @@
...
 
-Dask is a simple task scheduling system that uses directed acyclic graphs (DAGs)
-of tasks to break up large computations into many small ones.
+Dask is a flexible parallel computing library for analytics.  Dask emphasizes
+the following virtues:
 
-**Dask enables parallel computing** through task scheduling and blocked algorithms.
-This allows developers to write complex parallel algorithms and execute them
-in parallel either on a modern multi-core machine or on a distributed cluster.
+*  **Familiar**: Provides parallelized NumPy array and Pandas DataFrame objects
```
Comment (koverholt) [inline docs/source/index.rst]: It's good to mention Dask dataframes and arrays here, is there somewhere else in this topmost list to refer to bags and imperative?

File: docs/source/index.rst [hunk 4]
```diff
@@ -2,54 +2,101 @@
...
+Scales from laptops to clusters
+-------------------------------
+
+Dask is convenient on a laptop.  It :doc:`installs <install>` trivially with
+``conda`` or ``pip`` and extends the size of convenient datasets from "fits in
+memory" to "fits on disk".
+
+Dask can scale to a cluster of 100s of machines. It is resilient, elastic, data
+local, and low latency.  For more information see documentation on the
+`distributed scheduler`_
```
Comment (koverholt) [inline docs/source/index.rst]: Missing period.

File: docs/source/index.rst [hunk 5]
```diff
@@ -2,54 +2,101 @@
...
+
+**Dask Bag** mimics iterators, Toolz, PySpark
+
+.. code-block:: python
+
+   import dask.bag as db
+   b = db.from_filenames('2015-*-*.json.gz').map(json.loads)
+   b.pluck('name').frequencies().topk(10, lambda pair: pair[1]).compute()
+
+**Dask Imperative** mimics for loops
```
Comment (koverholt) [inline docs/source/index.rst]: Maybe "mimics for loops, can be used to {wrap or parallelize} custom code"?

File: docs/source/index.rst (hunk 3 above)
Comment (mrocklin) [inline docs/source/index.rst]: I've looked again and can't find a good spot for them.  I'm not overly concerned here.  It's hard for bag/imperative to convey much information before they're explained.  They're also included in the next section.

File: docs/source/index.rst (hunk 4 above)
Comment (mrocklin) [inline docs/source/index.rst]: Fixed

File: docs/source/index.rst (hunk 5 above)
Comment (mrocklin) [inline docs/source/index.rst]: Fixed.  Thanks!

**Pull request merged**